File: app/src/main/java/com/antarikshc/intrack/data/InvDbHelper.java
package com.antarikshc.intrack.data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.antarikshc.intrack.data.InvContract.InvEntry;

public class InvDbHelper extends SQLiteOpenHelper {

    public static final String LOG_TAG = InvDbHelper.class.getSimpleName();

    /**
     * Name and version of database
     **/
    private static final String DATABASE_NAME = "inventory.db";
    private static final int DATABASE_VERSION = 1;

    public InvDbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    /**
     * This is called when database is first created
     **/
    @Override
    public void onCreate(SQLiteDatabase db) {

        //String that contains the SQL statement to create the invent table
        String SQL_CREATE_INVENT_TABLE = "CREATE TABLE " + InvEntry.TABLE_NAME + " ("
                + InvEntry._ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                + InvEntry.COLUMN_ITEM_NAME + " TEXT NOT NULL, "
                + InvEntry.COLUMN_ITEM_STOCK + " INTEGER NOT NULL DEFAULT 0, "
                + InvEntry.COLUMN_ITEM_CAPACITY + " INTEGER, "
                + InvEntry.COLUMN_ITEM_ICON + " BLOB, "
                + InvEntry.COLUMN_ITEM_SUP_PHONE + " INTEGER, "
                + InvEntry.COLUMN_ITEM_SUP_EMAIL + " TEXT);";

        // Execute the SQL statement
        db.execSQL(SQL_CREATE_INVENT_TABLE);

    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

        //Delete the current database when upgrading

    }
}

File: app/src/main/java/com/antarikshc/intrack/data/InvDbHelper.kt
package com.antarikshc.intrack.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_CAPACITY
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_ICON
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_NAME
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_STOCK
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_SUP_EMAIL
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.COLUMN_ITEM_SUP_PHONE
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion.TABLE_NAME
import com.antarikshc.intrack.data.InvContract.InvEntry.Companion._ID

class InvDbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    /**
     * This is called when database is first created
     */
    override fun onCreate(db: SQLiteDatabase) {

        //String that contains the SQL statement to create the invent table
        val SQL_CREATE_INVENT_TABLE = ("CREATE TABLE " + TABLE_NAME + " ("
                + _ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                + COLUMN_ITEM_NAME + " TEXT NOT NULL, "
                + COLUMN_ITEM_STOCK + " INTEGER NOT NULL DEFAULT 0, "
                + COLUMN_ITEM_CAPACITY + " INTEGER, "
                + COLUMN_ITEM_ICON + " BLOB, "
                + COLUMN_ITEM_SUP_PHONE + " TEXT, " //TEXT because we will save country code too

                + COLUMN_ITEM_SUP_EMAIL + " TEXT);")

        // Execute the SQL statement
        db.execSQL(SQL_CREATE_INVENT_TABLE)

    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {

        //Delete the current database when upgrading

    }

    companion object {

        val LOG_TAG = InvDbHelper::class.java.simpleName

        /**
         * Name and version of database
         */
        private const val DATABASE_NAME = "inventory.db"
        private const val DATABASE_VERSION = 1
    }
}
